### PR TITLE
Bump golang to 1.19.2

### PR DESCRIPTION
**Pull request by bourgeoisor**
This PR bumps **all** Go containers to use `1.19.2` (latest stable):

- direct-to-sd: `1.15` -> `1.19.2`
- prometheus-to-sd: `1.12` -> `1.19.2`
- hello-app-cdn: `1.8` -> `1.19.2`
- hello-app-redis: `1.16` -> `1.19.2`
- hello-app-tls: `1.8` -> `1.19.2`
- hello-app: `1.8` -> `1.19.2`
- quickstart/go: `1.12` -> `1.19.2`
- wi-secrets: `1.18` -> `1.19.2`
- workload-metrics: `1.16` -> `1.19.2`

In addition, this PR also converges **all** Go containers to use the same builder pattern and increases consistency throughout the various Dockerfiles, as well as substitute uses of `alpine` to `gcr.io/distroless/base-debian11`:
```
FROM golang:1.19.2 as builder
WORKDIR /app
COPY * ./
RUN CGO_ENABLED=0 GOOS=linux go build -o /my-app-name

FROM gcr.io/distroless/base-debian11
WORKDIR /
COPY --from=builder /my-app-name /my-app-name
USER nonroot:nonroot
CMD ["/my-app-name"]
```

### CODE REVIEW

File: custom-metrics-autoscaling/direct-to-sd/Dockerfile [hunk 1]
```diff
@@ -12,12 +12,14 @@
...
-RUN go get cloud.google.com/go/monitoring/apiv3
-RUN CGO_ENABLED=0 GOOS=linux go install sd-dummy-exporter
+FROM golang:1.19.2 as builder
+WORKDIR /app
+COPY * ./
+RUN CGO_ENABLED=0 GOOS=linux go build -o /app
 
-FROM gcr.io/distroless/static:latest
-COPY --from=0 /go/bin/sd-dummy-exporter .
+FROM alpine
```
Comment (donmccasland) [inline custom-metrics-autoscaling/direct-to-sd/Dockerfile:20]: Is there another option here than alpine?  using alpine as a mixin is a surefire way to add security issues.

File: custom-metrics-autoscaling/prometheus-to-sd/Dockerfile [hunk 2]
```diff
@@ -12,10 +12,14 @@
...
-RUN go get github.com/prometheus/client_golang/prometheus
-RUN CGO_ENABLED=0 GOOS=linux go install prometheus-dummy-exporter
+FROM golang:1.19.2 as builder
+WORKDIR /app
+COPY * ./
+RUN CGO_ENABLED=0 GOOS=linux go build -o /app
 
-FROM gcr.io/distroless/static:latest
-COPY --from=0 /go/bin/prometheus-dummy-exporter .
+FROM alpine
```
Comment (donmccasland) [inline custom-metrics-autoscaling/prometheus-to-sd/Dockerfile:20]: same question as above

File: hello-app-cdn/Dockerfile [hunk 3]
```diff
@@ -1,8 +1,26 @@
...
 
-FROM alpine:latest
-COPY --from=0 /go/bin/hello-app-cdn .
+FROM golang:1.19.2 as builder
+WORKDIR /app
+RUN go mod init hello-app-cdn
+COPY *.go ./
+RUN CGO_ENABLED=0 GOOS=linux go build -o /app
+
+FROM alpine
```
Comment (donmccasland) [inline hello-app-cdn/Dockerfile:21]: same question

File: custom-metrics-autoscaling/direct-to-sd/Dockerfile [hunk 4]
```diff
@@ -12,12 +12,14 @@
...
+WORKDIR /
+RUN apk add --no-cache ca-certificates
```
Comment (donmccasland) [inline custom-metrics-autoscaling/direct-to-sd/Dockerfile:22]: I don't what what the alpine equivalent of apt-get update is, but we should make sure to do it before installing ca-certificates.

**Review by donmccasland — APPROVED** [review]

File: custom-metrics-autoscaling/direct-to-sd/Dockerfile (hunk 1 above)
Comment (bourgeoisor) [reply custom-metrics-autoscaling/direct-to-sd/Dockerfile]: Changed to distroless.

**Pull request merged**